internal/grpc/services/appprovider/appprovider.go

```diff
@@ -48,8 +49,9 @@ func init() {
 }
 
 type service struct {
-	provider app.Provider
-	conf     *config
+	provider   app.Provider
+	conf       *config
+	appsURLMap map[string]interface{}
```

Comment:
This needs to be protected by a lock. Multiple threads can try to fill this at the same time, and even though they would probably be writing the same data, there's no ordering in a map, so we might end up with garbage values.

Reply:
Well spotted, too much Python coding ;-) I'll fix that.

Reply:
You can consider using a sync.Map like done [here](https://github.com/cs3org/reva/blob/master/pkg/ocm/provider/authorizer/json/json.go#L83).

Reply:
In fact, given the usage (write once, read many), for now I went for a standard `mutex.Lock()` approach. Just committed.

Reply:
(true that once a regular refresh is implemented, the best option is a sync.Map)

Reply:
This won't work. The lock is local to each goroutine, so it won't know what it actually has to take a lock on. Ref: https://play.golang.org/p/RZfwVALEvRn. You'd have to add this lock to the service struct itself.

And why not use sync.Map in the first place since we agree it would be the better option in the long run?

Reply:
(Thanks for your patience, I genuinely assumed Go had some higher level constructs, like mutexes being stored in some common memory. Instead, it's pretty much like C/C++, good to know)

OK, I gave it a try with sync.Map but it's not a good fit as it's not a real map; and for the long run, we want to invalidate its content and call that function again to fill it. So at that point I'll implement a proper multi-read/single-write lock for it, but for this PR, I came back to all local variables. I guess it's better to move on for now and leave that as a separate task.